api/src/main/java/org/apache/brooklyn/api/location/LocationRegistry.java

```diff
@@ -86,25 +94,62 @@
     public Location resolve(String spec, @Nullable Map locationFlags);
     
     /**
-     * As {@link #resolve(String)} but takes collections (of strings or locations)
+     * As {@link #getLocationManaged(String)} but takes collections (of strings or locations)
      * <p>
      * Expects a collection of elements being individual location spec strings or locations, 
      * and returns a list of resolved (newly created and managed) locations.
      * <p>
      * From 0.7.0 this no longer flattens lists (nested lists are disallowed) 
      * or parses comma-separated elements (they are resolved as-is)
-     */
+     * @deprecated since 0.9.0 use {@link #getListOfLocationsManaged(Object)} */
+    @Deprecated
     public List<Location> resolve(Iterable<?> spec);
     
     /** Takes a string, interpreted as a comma-separated (or JSON style, when you need internal double quotes or commas) list;
      * or a list, passed to {@link #resolve(Iterable)}; or null/empty (empty list),
-     * and returns a list of resolved (created and managed) locations */
+     * and returns a list of resolved (created and managed) locations 
+     * @deprecated since 0.9.0 use {@link #getListOfLocationsManaged(Object)} */
+    @Deprecated
     public List<Location> resolveList(Object specList);
+
+    /** Create a {@link LocationSpec} representing the given spec string such as a named location 
+     * or using a resolver prefix such as jclouds:aws-ec2. 
+     * This can then be inspected, assigned to an {@link EntitySpec}, 
+     * or passed to {@link LocationManager#createLocation(LocationSpec)} to create directly.
+     * (For that last case, common in tests, see {@link #getLocationManaged(String)}.) */
+    public Maybe<LocationSpec<? extends Location>> getLocationSpec(String spec);
+    /** As {@link #getLocationSpec(String)} but also setting the given flags configured on the resulting spec. */
+    public Maybe<LocationSpec<? extends Location>> getLocationSpec(String spec, Map<?,?> locationFlags);
+
+    /** As {@link #getLocationSpec(String)} where the caller has a {@link LocationDefinition}. */
+    public Maybe<LocationSpec<? extends Location>> getLocationSpec(LocationDefinition ld);
+    /** As {@link #getLocationSpec(String,Map)} where the caller has a {@link LocationDefinition}. */
+    public Maybe<LocationSpec<? extends Location>> getLocationSpec(LocationDefinition ld, Map<?,?> locationFlags);
     
+    /** A combination of {@link #getLocationSpec(String)} then {@link LocationManager#createLocation(LocationSpec)},
+     * mainly for use in tests or specialised situations where a managed location is needed directly.
+     * The caller is responsible for ensuring that the resulting {@link Location} 
+     * is cleaned up, ie removed from management via {@link LocationManager#unmanage(Location)} directly or linking it to 
+     * an {@link Entity} or another {@link Location} which will unmanage it. */
+    public Location getLocationManaged(String spec);
```

Comment:
Personal preference to not mixing responsibilities. Now that this API is dealing with specs better leave it to utility classes to do the delegation to `LocationManager`. Same for following methods.